pytest_xpara/plugin.py

```diff
@@ -3,19 +3,35 @@
 import py.path
 
 
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers", "xparametrize: mark test to run only on named environment"
+    )
+
+
 def pytest_addoption(parser):
     group = parser.getgroup('xpara', 'extended parametrizing plugin')
     group.addoption('--xpara', action='store_true', default=False,
                     help='Enable the extended parametrizing support. '
-                    'default: False')
+                         'default: False')
 
 
 def pytest_generate_tests(metafunc):
     if not metafunc.config.option.xpara:
         return
 
     try:
-        mark = metafunc.function.xparametrize
+        if hasattr(metafunc.function, "xparametrize"):
+            mark = metafunc.function.xparametrize
+        elif hasattr(metafunc.function, "pytestmark"):
+            temp_marks = [x for x in metafunc.function.pytestmark
+                          if x.name == "xparametrize"]
+            if len(temp_marks) > 0:
+                mark = temp_marks[0]
+            else:
+                return
+        else:
+            return
     except AttributeError:
```

Comment:
Is this try-except block still necessary?

Reply:
I think one try-except block is enough